test/jdk/javax/swing/JFrame/bug4419914.java

```diff
@@ -28,55 +28,143 @@
+        """;
 
     public static void main(String[] args) throws Exception {
         PassFailJFrame.builder()
                 .title("Tab movement Instructions")
                 .instructions(INSTRUCTIONS)
                 .rows((int) INSTRUCTIONS.lines().count() + 2)
-                .columns(48)
-                .testUI(bug4419914::createTestUI)
+                .columns(45)
+                .testTimeOut(15)
+                .testUI(bug4419914::createAndShowUI)
+                .positionTestUI(WindowLayouts::rightOneRow)
                 .build()
                 .awaitAndCheck();
     }
 
-    private static JFrame createTestUI() {
-        JFrame frame = new JFrame("bug4419914");
+    private static List<Window> createAndShowUI() {
+        return List.of(createJFrame(), createJWindow(), createJDialog());
+    }
+
+    private static JFrame createJFrame() {
+        frame = new JFrame("bug4419914 JFrame");
         frame.setFocusCycleRoot(true);
+        // Tab movement set to RTL
         frame.setComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
         frame.setLocale(Locale.ENGLISH);
         frame.enableInputMethods(false);
 
+        // Component placement within content pane set to RTL
         frame.getContentPane().setComponentOrientation(
-                               ComponentOrientation.RIGHT_TO_LEFT);
+                ComponentOrientation.RIGHT_TO_LEFT);
         frame.getContentPane().setLocale(Locale.ENGLISH);
-        frame.getContentPane().setLayout(new BorderLayout());
+        frame.setLayout(new BorderLayout());
         frame.add(new JButton("SOUTH"), BorderLayout.SOUTH);
         frame.add(new JButton("CENTER"), BorderLayout.CENTER);
         frame.add(new JButton("END"), BorderLayout.LINE_END);
         frame.add(new JButton("START"), BorderLayout.LINE_START);
         frame.add(new JButton("NORTH"), BorderLayout.NORTH);
-        frame.setSize(300, 150);
+        frame.setSize(300, 160);
         return frame;
     }
+
+    private static JWindow createJWindow() {
+        JWindow window = new JWindow(frame);
+        window.setFocusableWindowState(true);
+        // Tab movement set to RTL
+        window.setComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+        window.setLocale(new Locale("en"));
+        window.enableInputMethods(false);
+
+        // Component placement within content pane set to RTL
+        window.getContentPane().setComponentOrientation(
+                ComponentOrientation.RIGHT_TO_LEFT);
+        window.getContentPane().setLocale(new Locale("en"));
+        window.setLayout(new BorderLayout());
+        window.add(new JLabel("bug4419914 JWindow"), BorderLayout.NORTH);
+        window.add(new JButton("START"), BorderLayout.LINE_START);
+        window.add(new JButton("CENTER"), BorderLayout.CENTER);
+        window.add(new JButton("END"), BorderLayout.LINE_END);
+
+        JButton quitButton = new JButton("Quit");
```

Comment:
For consistency with other UI component's text
```suggestion
        JButton quitButton = new JButton("QUIT");
```